dsl/src/main/scala/org/http4s/dsl/impl/ResponseGenerator.scala

```diff
@@ -37,10 +42,10 @@ trait EntityResponseGenerator[F[_]] extends Any with ResponseGenerator {
   def apply()(implicit F: Applicative[F]): F[Response[F]] =
     F.pure(Response(status, headers = Headers(`Content-Length`.zero)))
 
-  def apply[A](body: A)(implicit F: Monad[F], w: EntityEncoder[F, A]): F[Response[F]] =
-    apply(body, Headers.empty)(F, w)
+  def apply(headers: Header*)(implicit F: Applicative[F]): F[Response[F]] =
```

Comment:
Also doesn't need to be an overload.

Note: we don't extend `EmptyResponseGenerator` because we can (and should) set `Content-Length: 0` here.  We can't safely do it on the empty responses, because HTTP is a complicated beast.